pkg/auth/requests/filter.go

```diff
@@ -27,10 +28,13 @@ func (h authHeaderHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 		return
 	}
 
-	// clean extra
+	//clean extra that is not part of userInfo
 	for header := range req.Header {
 		if strings.HasPrefix(header, "Impersonate-Extra-") {
-			req.Header.Del(header)
+			key := strings.TrimPrefix(header, "Impersonate-Extra-")
+			if _, ok := userInfo.GetExtra()[key]; !ok {
```

Comment:
I don't think we can rely on the data that came from the request to do the filtering here, we need a list of ones we accept

Reply:
Added a static map to list the currently supported attributes